src/main/java/core/InputHandler.java

```diff
@@ -254,9 +264,10 @@ private void handleDelete(String inputData, OutputFormatter outputFormatter) {
             } else {
                 toBeDeleted = this.taskList.deleteTask(indexRetriever[1]);
             }
-            outputFormatter.appendAll("Noted. I've removed this task:", "\n", toBeDeleted, "\n", "Now you have ", taskList.getLength(), " task(s) in the list.");
+            outputFormatter.appendAll("Noted. I've removed this task:", "\n", toBeDeleted, "\n",
+                    "Now you have ", taskList.getLength(), " task(s) in the list.");
 
-        } catch ( NumberFormatException e) {
+        } catch (NumberFormatException e) {
```

Comment:
Before I read anymore of your code, I realized how much I am lacking in terms of exception handling. I essentially lumped all exceptions under a DukeException class, except that I modified the exception message to describe the specific reason the exception was thrown. In hindsight, I probably should have created individual Exception subclasses to handle each type of exception.